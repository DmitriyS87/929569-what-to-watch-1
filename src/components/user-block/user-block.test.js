import React from 'react';
import renderer from 'react-test-renderer';
import UserBlock from '../user-block/user-block.jsx';

it(`UserBlock correctly renders with logouted user: `, () => {
  const mockHandler = jest.fn();
  const tree = renderer.create(
      <UserBlock
        loginUser={mockHandler}
        user={null}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`UserBlock correctly renders with logined user: `, () => {
  const mockHandler = jest.fn();
  const tree = renderer.create(
      <UserBlock
        loginUser={mockHandler}
        user={{id: `mock`}}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
