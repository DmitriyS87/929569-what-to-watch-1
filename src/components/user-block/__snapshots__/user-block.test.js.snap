// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock correctly renders with logined user:  1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;

exports[`UserBlock correctly renders with logouted user:  1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="#"
    onClick={[MockFunction]}
  >
    Sign in
  </a>
</div>
`;
